python/cuml/tests/dask/test_dask_logistic_regression.py

```diff
@@ -924,3 +921,102 @@ def test_standardization_example(fit_intercept, regularization, client):
 
     assert array_equal(lron_coef_origin, sg.coef_)
     assert array_equal(lron_intercept_origin, sg.intercept_)
+
+
+@pytest.mark.mg
+@pytest.mark.parametrize("fit_intercept", [True, False])
+@pytest.mark.parametrize(
+    "regularization",
+    [
+        ("none", 1.0, None),
+        ("l2", 2.0, None),
+        ("l1", 2.0, None),
+        ("elasticnet", 2.0, 0.2),
+    ],
+)
+def test_standardization_sparse(fit_intercept, regularization, client):
+    n_rows = 10000
+    n_cols = 25
+    n_info = 15
+    n_classes = 4
+    nnz = int(n_rows * n_cols * 0.3)  # number of non-zero values
+    tolerance = 0.005
+
+    datatype = np.float32
+    n_parts = 10
+    max_iter = 5  # cannot set this too large. Observed GPU-specific coefficients when objective converges at 0.
+
+    penalty = regularization[0]
+    C = regularization[1]
+    l1_ratio = regularization[2]
+
+    est_params = {
+        "penalty": penalty,
+        "C": C,
+        "l1_ratio": l1_ratio,
+        "fit_intercept": fit_intercept,
+        "max_iter": max_iter,
+    }
+
+    def make_classification_with_nnz(
+        datatype, n_rows, n_cols, n_info, n_classes, nnz
+    ):
+        assert n_rows * n_cols >= nnz
+
+        X, y = make_classification_dataset(
+            datatype, n_rows, n_cols, n_info, n_classes=n_classes
+        )
+        X = X.flatten()
+        num_zero = len(X) - nnz
+        zero_indices = np.random.choice(
+            a=range(len(X)), size=num_zero, replace=False
+        )
+        X[zero_indices] = 0
+        X_res = X.reshape(n_rows, n_cols)
+        return X_res, y
+
+    X_origin, y = make_classification_with_nnz(
+        datatype, n_rows, n_cols, n_info, n_classes, nnz
+    )
+    X = csr_matrix(X_origin)
+    assert X.nnz == nnz and X.shape == (n_rows, n_cols)
+
+    from sklearn.preprocessing import StandardScaler
+
+    scaler = StandardScaler(with_mean=fit_intercept, with_std=True)
+    scaler.fit(X_origin)
+    scaler.scale_ = np.sqrt(scaler.var_ * len(X_origin) / (len(X_origin) - 1))
+    X_scaled = scaler.transform(X_origin)
+
+    X_da, y_da = _prep_training_data_sparse(
+        client, X, y, partitions_per_worker=n_parts
+    )
+    from cuml.dask.linear_model import LogisticRegression as cumlLBFGS_dask
+
+    assert X_da.shape == (n_rows, n_cols)
+
+    computed_csr = X_da.compute()
+    assert isinstance(computed_csr, csr_matrix)
+    assert computed_csr.nnz == nnz and computed_csr.shape == (n_rows, n_cols)
+    assert array_equal(computed_csr.data, X.data)
+    assert array_equal(computed_csr.indices, X.indices)
+    assert array_equal(computed_csr.indptr, X.indptr)
```

Comment:
Would be good to use the `tolerance` value consistently here, like the changes in https://github.com/rapidsai/cuml/blob/88d659ebb76a9d6b235842d9efb434a89af4336a/python/cuml/tests/dask/test_dask_logistic_regression.py#L400